src/util/plotting.jl

```diff
@@ -115,18 +117,20 @@ SamplePlot((f,)::Tuple{<:FiniteGP}) = SamplePlot((f.x, f))
 SamplePlot((x, gp)::Tuple{<:AbstractVector,<:AbstractGP}) = SamplePlot((gp(x, 1e-9),))
 
 @recipe function f(sp::SamplePlot)
-    nsamples::Int = get(plotattributes, :samples, 1)
+    nsamples::Int = pop!(plotattributes, :samples, 1)
     samples = rand(sp.f, nsamples)
 
+    col_label = pop!(plotattributes, :label, "")
+    if col_label isa String
+        # blank labels for all but first column, e.g. ["samples" "" "" ...]
+        col_label = [col_label fill("", nsamples - 1)...]
+    end
```

Comment:
It is better to avoid splatting and to use `hcat`. Also sometimes the compiler becomes confused if one reassigns a variable with a different type.
```suggestion
    _label = pop!(plotattributes, :label, "")
    col_label = if _label isa String
        # blank labels for all but first column, e.g. ["samples" "" "" ...]
        hcat(_label, fill("", nsamples - 1))
    else
        _label
    end
```

Reply:
> It is better to avoid splatting and to use hcat.

For my understanding: What is better about that ?

(That's actually what I had first but then I thought you'd come and tell me that the shorter splatting would be more idiomatic so then changed it to that :joy:)

Reply:
Splatting puts more stress on the compiler. It's fine if you use e.g. small tuples but especially when you work with possibly longer array it is good to avoid it if you can.